src/plugins/flow.js

```diff
@@ -1341,6 +1340,12 @@ export default (superClass: Class<Parser>): Class<Parser> => class extends super
 
   parseAssignableListItemTypes(param: N.Pattern): N.Pattern {
     if (this.eat(tt.question)) {
+      if (param.type !== "Identifier") {
```

Comment:
Is this expected behavior in flow? Using the fixture from the matching typescript test in astexplorer seems to produce a [funky result](https://astexplorer.net/#/gist/b55806bfc15230ce5df73f09335bdaa5/bdcd344d7137134d11afa84271176aa43d87bea7).

If it is, we should probably add a matching flow test fixture.